instrumentation/netty/netty-4.1/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/v4_1/client/HttpClientRequestTracingHandler.java

```diff
@@ -24,7 +24,7 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise prm) {
       return;
     }
 
-    Context parentContext = ctx.channel().attr(AttributeKeys.CONNECT_CONTEXT).getAndRemove();
+    Context parentContext = ctx.channel().attr(AttributeKeys.WRITE_CONTEXT).getAndRemove();
```

Comment:
Can you add some documentation maybe in `AttributeKeys` on when we'd parent on the connect context vs the write context? Or do we really need two keys?

Reply:
connect context is used in one specific use-case: when we create a span for connect failure. I plan to follow up with a separate PR that makes that use-case more obvious in the code. Is it Ok if I add documentation then, if it will still be needed?